codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolMetadataProvider.kt

```diff
@@ -67,10 +67,18 @@ abstract class SymbolMetadataProvider(private val base: RustSymbolProvider) : Wr
     abstract fun enumMeta(stringShape: StringShape): RustMetadata
 }
 
-class BaseSymbolMetadataProvider(base: RustSymbolProvider) : SymbolMetadataProvider(base) {
+/**
+ * The base metadata supports a list of attributes that are used by generators to decorate code.
+ * By default we apply #[non_exhaustive] only to client structures since model changes should
+ * be considered as breaking only when generating server code.
+ */
+class BaseSymbolMetadataProvider(
+    base: RustSymbolProvider,
+    additionalAttributes: List<Attribute> = listOf(NonExhaustive)
```

Comment:
rather than having this be the default, lets set it explicitly in `RustCodegenPlugin` in the client and remove the default